pkg/drivers/vfkit/vfkit.go

```diff
@@ -136,6 +136,49 @@ func (d *Driver) GetIP() (string, error) {
 	return d.IPAddress, nil
 }
 
+func writePidfile(pidfile string, pid int) error {
+	data := fmt.Sprintf("%v", pid)
+	return os.WriteFile(pidfile, []byte(data), 0600)
+}
+
+func readPidfile(pidfile string) (int, error) {
+	data, err := os.ReadFile(pidfile)
+	if err != nil {
+		return -1, err
+	}
+	pid, err := strconv.Atoi(strings.TrimSpace(string(data)))
+	if err != nil {
+		return -1, err
+	}
+	return pid, nil
+}
+
+func signalPidfile(pidfile string, sig syscall.Signal) error {
+	pid, err := readPidfile(pidfile)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+		// Already stopped.
+		os.Remove(pidfile)
+		return nil
+	}
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		return err
+	}
+	log.Infof("Sending signal %q to pid %v", sig, pid)
```

Comment:
`crc` checks that the process name contains `vfkit` in case the pid file has stale content, and the pid was reused by some unrelated process.

Reply:
It is https://github.com/crc-org/crc/blob/4bdcc16077a84583ad85ea5b8b75eefab13ba82d/pkg/drivers/vfkit/driver_darwin.go#L420-L438

Reply:
This is implemented now using the process package #20528.